src/main/java/org/dataone/portal/TokenGenerator.java

```diff
@@ -181,35 +184,52 @@ public String getJWT(String userId, String fullName)
 
         CertificateManager cmInst = CertificateManager.getInstance();
         // consumers do not need the private key
         if (privateKeyFileName != null) {
-            privateKey = (RSAPrivateKey) cmInst.loadPrivateKeyFromFile(privateKeyFileName,
-                                                                       privateKeyPassword);
+            // uses a null password:
+            privateKey = (RSAPrivateKey) cmInst.loadPrivateKeyFromFile(privateKeyFileName, null);
         }
     }
 
     /*
      * Set the public key
      * @throws IOException
      */
-    private void setPublicKey() throws IOException {
-        // use either the configured certificate, or fetch it from the CN
-        String certificateFileName =
-            Settings.getConfiguration().getString("cn.server.publiccert.filename");
+    protected void setPublicKeys() throws IOException {
+
+        publicKeys = new ArrayList<>();
+        List<BigInteger> publicKeyModuli = new ArrayList<>();
+
+        // use the configured certificate, if it exists, and also the one fetched from the CN
+        String[] certificateFileNames =
+            Settings.getConfiguration().getStringArray("cn.server.publiccert.filename");
         CertificateManager cmInst = CertificateManager.getInstance();
-        log.debug("certificateFileName=" + certificateFileName);
-        if (certificateFileName != null && certificateFileName.length() > 0) {
-            publicKey =
-                (RSAPublicKey) cmInst.loadCertificateFromFile(certificateFileName).getPublicKey();
-        } else {
-            Certificate cert = fetchServerCertificate();
-            log.debug("using certificate from server: " + cert);
-            if (cert != null) {
-                publicKey = (RSAPublicKey) cert.getPublicKey();
-            }  // what happens if publicKey is null?
+        log.debug("certificateFileNames: \n" + Arrays.toString(certificateFileNames));
+        for (String certFileName : certificateFileNames) {
+            if (!Files.isReadable(Paths.get(certFileName))) {
+                log.warn("Certificate file " + certFileName + " does not exist.");
+                continue;
+            }
+            RSAPublicKey currentKey =
+                (RSAPublicKey) cmInst.loadCertificateFromFile(certFileName).getPublicKey();
+            BigInteger currentKeyModulus = currentKey.getModulus();
+            if (publicKeyModuli.contains(currentKeyModulus)) {
+                log.warn("Certificate file " + certFileName + " is a duplicate.");
+                continue;
+            }
+            publicKeys.add(currentKey);
+            publicKeyModuli.add(currentKeyModulus);
+        }
+
+        Certificate cert = fetchServerCertificate();
+        log.debug("using certificate from server: " + cert);
+        if (cert != null) {
+            // keep a copy, so we can periodically check if it's been updated...
+            serverPublicKey = (RSAPublicKey) cert.getPublicKey();
```

Comment:
Do we need to check if there is a same public key already in the list before we add the one from the cn server url? In theory, the list can have one.

Reply:
done 2d638c7a2c57ae4e9ef92c6b114c33716807ee53